### DISCUSSION TITLE
Allow for a default Player to be selected

### DISCUSSION THREAD

**Discussion by tjharman** [music-assistant/support#2328] [Feature requests and ideas]
Every time I open up Music Assistant I have to select the player I want.  For me personally, 99% of the time this is the same player for me, the kitchen speaker.

I'd love an option I can tick in Players that's "Make this Player the default selection" or similar, and if ticked, by default when Music Assistant is opened and would, at present say "No Player Selected" the Player with "Make this Player the default selection" ticked would be automatically selected.

I understand that having a default player selected was an option previously but that this then meant a number of users ended up having music playing on a device they didn't mean to.  Thus the default out-of-the-box experience should be as it is today, no default is selected.  My feature request only suggests providing the ability to select (And it should also be possible to untick the default, reverting to the current behaviour)

Thank you very much for considering (or not!) this request.

Tim

Here is [my search](https://github.com/orgs/music-assistant/discussions?discussions_q=Default+Player) for a similar feature request.

**Comment by rxritalin** [comment]
If I understand what you are wanting, I think I have a better implementation idea. 
https://github.com/orgs/music-assistant/discussions/2346

If you use the built in UI for MA, then this wont really work. 
Personally I dont use it that often. I use the web-view interface more often. This includes on my phone and in the lovelace UI, via the iFrame card. In truth we can already use hashes, but it seems its only for navigating the available media. Its also not very intuitive, though it does not really need to be. 
For instance http://192.168.86.6:8095/#/home loads my home screen in the MA. 
But if I change the hash to  #/playlists it would load the playlists screen. 
Likewisie I could do #/playlists/library/64 which would load up my 3AM Airport playlist. 
What we need is the ability to add a # for the speaker or speaker group we want to use.

**Comment by Paul-Vdp** [comment]
Sounds like a no-brainer to me too (at least the OP's request).
I would be surprised if the majority of users does NOT use the same player most of the time ...

**Comment by marcelveldt** [comment]
Its actually pretty funny - we actually had this in the code, we simply remembered the last selected player.
But that lead to a lot of confusion because people didn't understand what music started playing somewhere etc. so we decided to actually remove this part for now and make it a considered user choice on what player the music should start playing instead of just assuming the last known player.

So its a whole lot more nuanced then people may think. We are slowly working our way to make the experience more consistent so when you first click an item or play button we will ask once for the player and then remember it. 
We may also just offer a setting somewhere to optionally bring back the old behavior of remembering the last player for folks that just have a small number of players or play to the same player all the time.

**Reply by chasut** [reply]
Either way (or both) you go, this would be very much appreciated.  But just to be clear, I certainly wouldn't want it to just automatically start playing, just not have to reselect the player every time.  It seems the queue is cleared as well so assume it wouldn't.  But having that remain as "last played" is also desirable  for me.
For me, both the "player(s)" and music source "queue" remain static from day to day.  I just want to walk up and hit play and enjoy.

**Reply by Thyraz** [reply]
I think main part of the confusion was, because on mobile devices you currently don't see the selected player name due to the small screen.

If the changes we discussed on Discord (player name above artist/song on mobile screen similar to the layout of the new Sonos app) are considered good after a beta testing, I guess there wouldn't be any confusion about that anymore, and remembering the last speaker might be just fine again. 

But sure, an option is always nice.

**Comment by Paul-Vdp** [comment]
Dare I say that with the current behaviour there's a lot of annoyance because people don't understand why they have to keep on choosing the speaker ?
Speaking for 'these people', I wholeheartedly would welcome the realisation of both your own suggestions : provide an (optional) 'default' setting AND a setting to 'let Mass manage my default speaker' (cfr the comparable setting in Windows regarding printers).
Thanks beforehand ;-)

**Comment by mzurhorst** [comment]
I would appreciate such a "default" as well since I only have one speaker (also in the kitchen  ;-) ).

**Comment by madbrain76** [comment]
My 2 cents - if you have one player, then it should be selected by default. In other cases, a default player will not work well. Unless I disable some that are different interfaces (Airplay + Chromecast + DLNA) to the same physical device, I have over 40 players showing up. It can take time to select the intended one, especially on a mobile device with a small screen.

In an ideal world, there could be automatic player selection based on smartphone/tablet location, with bluetooth beacons peppered around the house or backyard that the device could infer location from. I intend to explore this possibility with HA first now that there are conditional dashboards. If this works well enough, maybe a similar approach could be used for automatic player selection, optionally of course.

Sometimes, one still wants to start playback in a different zone, or a group, in which case manual selection remains possible.
Devices that don't have Bluetooth built-in, such as desktops, will be at a disadvantage when trying to locate the right player. Adding a bluetooth USB dongle is one possibility, but it needs to be made available to HA in the browser, since it isn't a native program. That may only work on some platforms/browsers.
There are also other ways to identify the location for fixed devices, either IP-based with a table if using DHCP reservations, or if not, MAC-based using reverse ARP, also with a table.
Or the player autoselection can simply be left out on devices which cannot reliably be located, of course.
I also have 3 rooms that have 2 sound systems in them (don't ask), so I would still need some manual selection for those.
With user location & player location, the player selection UI could also sort them by distance when selecting manually.

Note that I'm not saying any of this should be implemented right now, just throwing ideas out there.

**Reply by rxritalin** [reply]
Some of your thoughts are why I was leaning towards having a URL hash for speaker selection to be an option. 
This would give advanced users all the power they would need to trigger playback on specific speakers through scripts and automations.

**Comment by tjharman** [comment]
I'm not sure what's happened in the latest updates, but now when I open the app it seems to remember the last speaker I had selected.

This is great! I can just hit the music I want to listen to now and it plays where I'm hoping it'll play.

I haven't seen any notification/activity on this ticket, but it seems something has changed recently because now it's a much easier flow to get music playing.  I really appreciate that.  Thank you.

**Reply by OzGav** [reply]
Yes the last player is remembered now.

**Reply by tjharman** [reply]
Thanks @OzGav. I'm happy for this issue to be closed, unless you'd like to leave it open to discuss futher ideas.  Once again thanks for your hard work on this project, and for your time spent on this ticket.
